cpp/models/abm/location.cpp

```diff
@@ -36,12 +36,15 @@ Location::Location(LocationType type, uint32_t index, uint32_t num_cells)
     , m_index(index)
     , m_capacity(LocationCapacity())
     , m_capacity_adapted_transmission_risk(false)
-    , m_subpopulations{}
+    , m_subpopulations_time_series(Eigen::Index(InfectionState::Count))
     , m_cached_exposure_rate({AgeGroup::Count, VaccinationState::Count})
     , m_cells(std::vector<Cell>(num_cells))
     , m_required_mask(MaskType::Community)
     , m_npi_active(false)
 {
+    // Initialise the first time point and set the subpopulation values to 0. 
+    m_subpopulations_time_series.add_time_point();
+    m_subpopulations_time_series.get_last_value().setZero();
```

Comment:
Do we need it to be zero? Shouldn't it be the value of persons that are there at the beginning?

Reply:
It initialises the subpopulation time_series object so I set everything to zero, not the population. There're three functions "add_person", "change_state" and "change_subpopulation", which will update the last index of this object to the initial population.